File: SecondYear/semestrul4/MPP/PROIECT/CSharpProjectNetworking/Model/CategorieVarsta.cs
using System.Text.Json.Serialization;

namespace Model;

[Serializable]
public class CategorieVarsta : Entity<int>
{
    private int varstaMin;
    private int varstaMax;
        
    public CategorieVarsta(){}
    
    [JsonConstructor]
    public CategorieVarsta(int varstaMin, int varstaMax)
    {
        this.varstaMin = varstaMin;
        this.varstaMax = varstaMax;
    }
        
    public int VarstaMin
    {
        get { return varstaMin; }
        set { varstaMin = value; }
    }

    public override string ToString()
    {
        return
            $"{nameof(varstaMin)}: {varstaMin}, {nameof(varstaMax)}: {varstaMax}, {nameof(VarstaMin)}: {VarstaMin}, {nameof(VarstaMax)}: {VarstaMax}";
    }

    public int VarstaMax
    {
        get { return varstaMax; }
        set { varstaMax = value; }
    }
        
    public override bool Equals(object obj)
    {
        if (this == obj) return true;
        if (obj == null || GetType() != obj.GetType()) return false;
            
        CategorieVarsta that = (CategorieVarsta)obj;
        return varstaMin == that.varstaMin && varstaMax == that.varstaMax;
    }
        
    public override int GetHashCode()
    {
        return HashCode.Combine(varstaMin, varstaMax);
    }
}

File: SecondYear/semestrul4/MPP/PROIECT/JavaProjectORM/ProjectModel/src/main/java/ro/mpp2025/javaprojectui/CategorieVarsta.java
package ro.mpp2025.javaprojectui;

import java.util.Objects;

public class CategorieVarsta extends Entity<Integer>{

    private int varstaMin;
    private int varstaMax;

    public CategorieVarsta(int varstaMin, int varstaMax) {
        this.varstaMin = varstaMin;
        this.varstaMax = varstaMax;
    }

    public CategorieVarsta() {}

    public int getVarstaMin() {
        return varstaMin;
    }

    public int getVarstaMax() {
        return varstaMax;
    }

    public void setVarstaMax(int varstaMax) {
        this.varstaMax = varstaMax;
    }

    public void setVarstaMin(int varstaMin) {
        this.varstaMin = varstaMin;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        CategorieVarsta that = (CategorieVarsta) o;
        return varstaMin == that.varstaMin && varstaMax == that.varstaMax;
    }

    @Override
    public int hashCode() {
        return Objects.hash(varstaMin, varstaMax);
    }

    @Override
    public String toString() {
        return "CategorieVarsta{" +
                "varstaMin=" + varstaMin +
                ", varstaMax=" + varstaMax +
                '}';
    }
}
